pkg/sql/tests/sysbench_test.go

```diff
@@ -0,0 +1,624 @@
+			indexValue := s.indexValue(indexKey)
+			b.Put(pkKey, pkValue)
+			b.Put(indexKey, indexValue)
+		}
+		try0(s.db.Run(s.ctx, &b))
+	}
+
+	// Scan across all tables to ensure that intent resolution has completed.
+	for i := range sysbenchTables {
+		try(s.db.Scan(s.ctx, s.pkPrefix[i], s.indexPrefix[i].PrefixEnd(), 0))
+	}
+}
+
+func sysbenchExecutePointSelects(s sysbenchDriver, rng *rand.Rand) {
+	t := randTableNum(rng)
+	for range sysbenchPointSelects {
+		s.PointSelect(t, randRowID(rng))
+	}
+}
+
+func sysbenchExecuteRange(op func(tableNum, rowID, rowID), count int, rng *rand.Rand) {
+	t := randTableNum(rng)
+	for range count {
+		id := randRowID(rng)
+		id2 := id + sysbenchRangeSize - 1
+		op(t, id, id2)
+	}
+}
+
+func sysbenchExecuteSimpleRanges(s sysbenchDriver, rng *rand.Rand) {
+	sysbenchExecuteRange(s.SimpleRange, sysbenchSimpleRanges, rng)
+}
+
+func sysbenchExecuteSumRanges(s sysbenchDriver, rng *rand.Rand) {
+	sysbenchExecuteRange(s.SumRange, sysbenchSumRanges, rng)
+}
+
+func sysbenchExecuteOrderRanges(s sysbenchDriver, rng *rand.Rand) {
+	sysbenchExecuteRange(s.OrderRange, sysbenchOrderRanges, rng)
+}
+
+func sysbenchExecuteDistinctRanges(s sysbenchDriver, rng *rand.Rand) {
+	sysbenchExecuteRange(s.DistinctRange, sysbenchDistinctRanges, rng)
+}
+
+func sysbenchExecuteIndexUpdates(s sysbenchDriver, rng *rand.Rand) {
+	t := randTableNum(rng)
+	for range sysbenchIndexUpdates {
+		s.IndexUpdate(t, randRowID(rng))
+	}
+}
+
+func sysbenchExecuteNonIndexUpdates(s sysbenchDriver, rng *rand.Rand) {
+	t := randTableNum(rng)
+	for range sysbenchNonIndexUpdates {
+		s.NonIndexUpdate(t, randRowID(rng), randCValue(rng))
+	}
+}
+
+func sysbenchExecuteDeleteInserts(s sysbenchDriver, rng *rand.Rand) {
+	t := randTableNum(rng)
+	for range sysbenchDeleteInserts {
+		id := randRowID(rng)
+		k := randKValue(rng)
+		c := randCValue(rng)
+		pad := randPadValue(rng)
+		s.DeleteInsert(t, id, k, c, pad)
+	}
+}
+
+// sysbenchWorkload executes a single transaction of a sysbench workload.
+type sysbenchWorkload func(sysbenchDriver, *rand.Rand)
+
+func sysbenchOltpReadOnly(s sysbenchDriver, rng *rand.Rand) {
+	s.Begin()
+	sysbenchExecutePointSelects(s, rng)
+	sysbenchExecuteSimpleRanges(s, rng)
+	sysbenchExecuteSumRanges(s, rng)
+	sysbenchExecuteOrderRanges(s, rng)
+	sysbenchExecuteDistinctRanges(s, rng)
+	s.Commit()
+}
+
+func sysbenchOltpWriteOnly(s sysbenchDriver, rng *rand.Rand) {
+	s.Begin()
+	sysbenchExecuteIndexUpdates(s, rng)
+	sysbenchExecuteNonIndexUpdates(s, rng)
+	sysbenchExecuteDeleteInserts(s, rng)
+	s.Commit()
+}
+
+func sysbenchOltpReadWrite(s sysbenchDriver, rng *rand.Rand) {
+	s.Begin()
+	sysbenchExecutePointSelects(s, rng)
+	sysbenchExecuteSimpleRanges(s, rng)
+	sysbenchExecuteSumRanges(s, rng)
+	sysbenchExecuteOrderRanges(s, rng)
+	sysbenchExecuteDistinctRanges(s, rng)
+	sysbenchExecuteIndexUpdates(s, rng)
+	sysbenchExecuteNonIndexUpdates(s, rng)
+	sysbenchExecuteDeleteInserts(s, rng)
+	s.Commit()
+}
+
+func BenchmarkSysbench(b *testing.B) {
```

Comment:
Should we have a flavor that disables the local server optimization? Otherwise we're eliding a bunch of networking/encoding/decoding overhead here.